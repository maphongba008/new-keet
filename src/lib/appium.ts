import { ViewProps } from 'react-native'
import _includes from 'lodash/includes'

type NamedEnum<Key extends string> = { [v in Key]: v }

function inferGeneric<T extends string>(t: NamedEnum<T>) {
  return t
}

export const APPIUM_IDs = inferGeneric({
  back_button: 'back_button',
  // LobbyScreen
  lobby_btn_join_room: 'lobby_btn_join_room',
  lobby_btn_create_room: 'lobby_btn_create_room',
  lobby_btn_create_chat: 'lobby_btn_create_chat',
  lobby_btn_create_broadcast: 'lobby_btn_create_broadcast',
  lobby_btn_create_back: 'lobby_btn_create_back',
  lobby_input_room_name: 'lobby_input_room_name',
  lobby_input_room_description: 'lobby_input_room_description',
  lobby_btn_paste: 'lobby_btn_paste',
  lobby_btn_qr: 'lobby_btn_qr',
  lobby_btn_submit: 'lobby_btn_submit',
  lobby_btn_add_room: 'lobby_btn_add_room',
  lobby_tap_to_call: 'lobby_tap_to_call',
  lobby_input_search: 'lobby_input_search',
  lobby_btn_clear_search: 'lobby_btn_clear_search',
  lobby_btn_rooms: 'lobby_btn_rooms',
  lobby_btn_profile: 'lobby_btn_profile',
  lobby_dm: 'lobby_dm',
  lobby_filter: 'lobby_filter',
  lobby_selected_filter: 'lobby_selected_filter',
  lobby_joining_rooms: 'lobby_joining_rooms',
  lobby_swipable_btn_mute: 'lobby_swipable_btn_mute',
  lobby_swipable_btn_leave: 'lobby_swipable_btn_leave',
  lobby_bottomsheet_btn_leave: 'lobby_bottomsheet_btn_leave',
  // CreateBroadcastRoomScreen
  broadcast_accept_admin: 'broadcast_accept_admin',
  broadcast_btn_create: 'broadcast_btn_create',
  // RoomScreen
  room_call_join: 'room_call_join',
  room_call_mute: 'room_call_mute',
  room_call_video: 'room_call_video',
  room_call_speaker: 'room_call_speaker',
  room_call_end: 'room_call_end',
  room_btn_attach: 'room_btn_attach',
  room_btn_send: 'room_btn_send',
  room_input_msg: 'room_input_msg',
  room_attach_doc: 'room_attach_doc',
  room_attach_media: 'room_attach_media',
  room_capture_video: 'room_capture_video',
  room_capture_img: 'room_capture_img',
  room_btn_edit: 'room_btn_edit',
  room_btn_reply: 'room_btn_reply',
  room_btn_download: 'room_btn_download',
  room_btn_copy_msg: 'room_btn_copy_msg',
  room_btn_copy_msg_link: 'room_btn_copy_msg_link',
  room_btn_copy_img: 'room_btn_copy_img',
  room_btn_share: 'room_btn_share',
  room_btn_forward: 'room_btn_forward',
  room_btn_flag: 'room_btn_flag',
  room_btn_del: 'room_btn_del',
  room_btn_clear: 'room_btn_clear',
  room_btn_invite: 'room_btn_invite',
  room_btn_pin: 'room_btn_pin',
  room_profile_name: 'room_profile_name',
  room_profile_avatar: 'room_profile_avatar',
  room_pinned_message_header: 'room_pinned_message_header',
  // RoomOptionsScreen
  options_edit_details: 'options_edit_details',
  options_btn_files: 'options_btn_files',
  options_btn_share_link: 'options_btn_share_link',
  options_btn_share_qr: 'options_btn_share_qr',
  options_btn_clear_files: 'options_btn_clear_files',
  options_btn_leave: 'options_btn_leave',
  options_btn_toggle_notifications: 'options_btn_toggle_notifications',
  options_btn_toggle_call: 'options_btn_toggle_call',
  // InviteQRScreen
  invite_btn_type: 'invite_btn_type',
  invite_btn_expires_hours: 'invite_btn_expires_hours',
  invite_btn_expires_days: 'invite_btn_expires_days',
  invite_btn_expires_weeks: 'invite_btn_expires_weeks',
  invite_btn_no_expire: 'invite_btn_no_expire',
  invite_btn_accept_admin: 'invite_btn_accept_admin',
  invite_btn_generate: 'invite_btn_generate',
  // RoomMembersScreen
  members_search: 'members_search',
  // AccountScreen
  profile_pick_avatar: 'profile_pick_avatar',
  profile_edit_name: 'profile_edit_name',
  profile_network_status: 'profile_network_status',
  account_btn_share_network_info: 'account_btn_share_network_info',
  account_btn_copy_network_info: 'account_btn_copy_network_info',
  profile_tos: 'profile_tos',
  // list
  list_item: 'list_item_' as any,
  create_channel_btn: 'create_channel_btn',
  // Avatar
  avatar_image: 'avatar_image',
  // onboarding
  onboarding_btn_skip: 'onboarding_btn_skip',
  onboarding_btn_next: 'onboarding_btn_next',
  onboarding_btn_prev: 'onboarding_btn_prev',
  onboarding_btn_finish: 'onboarding_btn_finish',
  onboarding_tos_btn_let_in: 'onboarding_tos_btn_let_in',
  onboarding_input_username: 'onboarding_input_username',
  onboarding_btn_save_username: 'onboarding_btn_save_username',
  // IdentityScreen
  identity_btn_create: 'identity_btn_create',
  identity_btn_link: 'identity_btn_link',
  identity_btn_skip: 'identity_btn_skip',
  // Room Participant list
  participant_item: 'participant_item_' as any,
  participant_btn_all: 'participant_btn_all',
  // Tooltips
  tooltip_btn_profile: 'tooltip_btn_profile',
  tooltip_btn_join_room: 'tooltip_btn_join_room',
  // PasscodeScreen
  passcode_btn_back: 'passcode_btn_back',
  passcode_btn_close: 'passcode_btn_close',
  passcode_btn_delete: 'passcode_btn_delete',
  passcode_confirm_btn_back: 'passcode_confirm_btn_back',
  passcode_confirm_btn_close: 'passcode_confirm_btn_close',
  // RecoverAccountScreen
  recover_btn_link: 'recover_btn_link',
  recover_btn_recover: 'recover_btn_recover',
  recover_input_passphrase: 'recover_input_passphrase',
  recover_btn_continue: 'recover_btn_continue',
  recover_btn_paste: 'recover_btn_paste',
  recover_btn_success: 'recover_btn_success',
  // IdentitySyncScreen
  identity_accept_checkbox: 'identity_accept_checkbox',
  identity_btn_accept: 'identity_btn_accept',
  identity_btn_use_camera_action: 'identity_btn_use_camera_action',
  identity_btn_use_invite_link: 'identity_btn_use_invite_link',
  identity_sync_invite_link_input: 'identity_sync_invite_link_input',
  identity_sync_invite_link_submit: 'identity_sync_invite_link_submit',
  identity_sync_invite_link_declined: 'identity_sync_invite_link_declined',
  identity_sync_invite_request_confirm: 'identity_sync_invite_request_confirm',
  identity_sync_invite_request_notMyDevice:
    'identity_sync_invite_request_notMyDevice',
  identity_sync_invite_request_success: 'identity_sync_invite_request_success',
  // Create ID
  create_identity_agreement_btn: 'create_identity_agreement_btn',
  create_identity_view_phrase_btn: 'create_identity_view_phrase_btn',
  create_identity_copy_phrase_btn: 'create_identity_copy_phrase_btn',
  create_identity_setup_continue_btn: 'create_identity_setup_continue_btn',
  create_id_verification_word: 'create_id_verification_word',
  create_id_submit_verification: 'create_id_submit_verification',
  create_id_onFinish_btn: 'create_id_onFinish_btn',
  // Components
  radio_btn: 'radio_btn',
  radio_btn_checked: 'radio_btn_checked',
  // Call Screen
  call_shared_screen_expand_button: 'call_shared_screen_expand_button',
  // MediaPreviewScreen
  media_preview_video_view: 'media_preview_video_view',
  media_preview_btn_custom_play: 'media_preview_btn_custom_play',
  media_preview_loading: 'media_preview_loading',
  // Discover Community
  discover_community_join_room: 'discover_community_join_room',
  discover_community_select_room: 'discover_community_select_room',

  // wallet
  wallet_default_server_action: 'wallet_default_server_action',
  wallet_custom_server_action: 'wallet_custom_server_action',
  wallet_server_get_started: 'wallet_server_get_started',
  wallet_setup_onFinish_btn: 'wallet_setup_onFinish_btn',
  wallet_create_agreement_btn: 'wallet_create_agreement_btn',
  wallet_settings_option_any: 'wallet_settings_option_any',
  wallet_settings_option_custom: 'wallet_settings_option_custom',
  wallet_settings_amount_input: 'wallet_settings_amount_input',
  wallet_settings_save_btn: 'wallet_settings_save_btn',
  wallet_send_submit_btn: 'wallet_send_submit_btn',
  wallet_send_scanner_btn: 'wallet_send_scanner_btn',
  wallet_send_scanner_back_btn: 'wallet_send_scanner_back_btn',
  wallet_send_open_scanner_btn: 'wallet_send_open_scanner_btn',
  wallet_send_paste_btn: 'wallet_send_paste_btn',
  wallet_input_wallet_address: 'wallet_input_wallet_address',
  wallet_input_transaction_description: 'wallet_input_transaction_description',
  wallet_btn_transfer_next: 'wallet_btn_transfer_next',
  wallet_btn_select_keet_contact: 'wallet_btn_select_keet_contact',
  wallet_btn_select_crypto_wallet: 'wallet_btn_select_crypto_wallet',
  wallet_receive_generate_invoice_btn: 'wallet_receive_generate_invoice_btn',
  wallet_receive_copy_invoice_btn: 'wallet_receive_copy_invoice_btn',
  wallet_receive_share_invoice_btn: 'wallet_receive_share_invoice_btn',
  wallet_custom_server_set_mode_btn: 'wallet_custom_server_set_mode_btn',
  wallet_custom_server_continue_btn: 'wallet_custom_server_continue_btn',
  wallet_custom_server_host_input: 'wallet_custom_server_host_input',
  wallet_custom_server_port_input: 'wallet_custom_server_port_input',
  wallet_custom_server_eth_indexer_input:
    'wallet_custom_server_eth_indexer_input',
  wallet_custom_server_connect_btn: 'wallet_custom_server_connect_btn',
  // User Profile Screen
  user_profile_chat: 'user_profile_chat',
  user_profile_call: 'user_profile_call',
  user_profile_send_dm: 'user_profile_send_dm',
} as const)

const DYNAMIC_KEY = [APPIUM_IDs.list_item]

export const appiumTestProps = (
  id?: keyof typeof APPIUM_IDs | string,
): Pick<ViewProps, 'accessible' | 'testID' | 'accessibilityLabel'> => {
  try {
    if (!id) {
      return {}
    }

    const appId = APPIUM_IDs[id as keyof typeof APPIUM_IDs]
    if (appId)
      return { accessible: true, testID: appId, accessibilityLabel: appId }

    if (DYNAMIC_KEY.some((substring) => _includes(id, substring))) {
      return { accessible: true, testID: id, accessibilityLabel: id }
    }
    return {}
  } catch (e) {
    console.log(e)
    return {}
  }
}
