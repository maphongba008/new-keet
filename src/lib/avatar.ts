import { ViewStyle } from 'react-native'
import b4a from 'b4a'

import { DEFAULT_AVATAR_COLORS } from './commonStyles'

export const SEED_BITS = 32
export const EMOJI_REGEXP = /\p{Emoji}/u

export const name2seed = (name: string) => {
  if (name.length === SEED_BITS) {
    return name
  } else if (name.length > SEED_BITS) {
    return name.substring(0, SEED_BITS - 1)
  }
  return name.padEnd(SEED_BITS)
}

function randInt(min: number, max: number, seed?: string): number {
  const rand = seed
    ? b4a.readUInt32LE(b4a.from(seed, 'hex')) / (2 ** SEED_BITS - 1)
    : Math.random()
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(rand * (max - min)) + min
}

export const randomColor = (seed?: string) => {
  return DEFAULT_AVATAR_COLORS[randInt(0, DEFAULT_AVATAR_COLORS.length, seed)]
}

export function getScaledFontSize({
  width,
  height,
}: Pick<ViewStyle, 'width' | 'height'>): number {
  const baseWidth = 30
  const baseHeight = 30
  const baseFontSize = 12
  const maxFontThreshold = 40

  if (typeof height !== 'number' || typeof width !== 'number') {
    return baseFontSize
  }

  // Calculate scaling factor
  const scalingFactor = (width * height) / (baseWidth * baseHeight)

  // Scale font size
  const fontSize = baseFontSize * Math.sqrt(scalingFactor)
  const minThreshold = Math.max(fontSize, baseFontSize)

  return Math.min(minThreshold, maxFontThreshold)
}
