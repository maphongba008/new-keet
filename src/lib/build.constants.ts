import Config from 'react-native-config'

const PRODUCT = {
  // define flags in alphabetical order
  FORCE_UPGRADE_BANNER: true,
  GRADIENT_TAB_BAR_ICON: true,
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  IS_TIME_STATS_ENABLED: true,
  SHOW_ADMIN_RESET_UI: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: false,
  SHOW_FULLSCREEN_AD: false,
  SHOW_MESSAGE_COPY_LINK: false,
  SHOW_NEW_PROFILE_FEATURE: false,
  SHOW_REMOVE_DEVICE: false,
  SHOW_RENAME_DEVICE: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: false,
  SHOW_UPDATE_BANNER: true,
  SHOW_WALLET: false,
  TRANSPARENT_TAB_BAR: true,
}

// KEEP this eslint statement or will raise lint error when swith to other setup
const STAGING = {
  ...PRODUCT,
  // define flags in alphabetical order
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: true,
  SHOW_MESSAGE_COPY_LINK: true,
  SHOW_NEW_PROFILE_FEATURE: true,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: true,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_WALLET: true,
  SHOW_ADMIN_RESET_UI: true,
  SHOW_REMOVE_DEVICE: true,
  SHOW_RENAME_DEVICE: true,
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const SV = {
  ...PRODUCT,
  // define flags in alphabetical order
  GRADIENT_TAB_BAR_ICON: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_UPDATE_BANNER: false,
  TRANSPARENT_TAB_BAR: false,
  IS_CHAT_CACHE_ENABLED: true,
}

// Specify a feature set alphabetically
const APP = STAGING

export const FORCE_UPGRADE_BANNER = APP?.FORCE_UPGRADE_BANNER

export const GRADIENT_TAB_BAR_ICON = APP?.GRADIENT_TAB_BAR_ICON

export const IS_SENTRY_ENABLED = APP?.IS_SENTRY_ENABLED

export const SHOW_BROADCAST_FILTER = APP?.SHOW_BROADCAST_FILTER

export const SHOW_CHANNEL_UI = APP?.SHOW_CHANNEL_UI

export const SHOW_CLEAN_DEVICE_SCREEN = APP?.SHOW_CLEAN_DEVICE_SCREEN

export const SHOW_FULLSCREEN_AD = APP?.SHOW_FULLSCREEN_AD

export const SHOW_MESSAGE_COPY_LINK = APP?.SHOW_MESSAGE_COPY_LINK

export const SHOW_NEW_PROFILE_FEATURE = APP?.SHOW_NEW_PROFILE_FEATURE

export const SHOW_PARTICIPANT_SHEET_USER_NAME =
  APP?.SHOW_PARTICIPANT_SHEET_USER_NAME

export const SHOW_PASSCODE = APP?.SHOW_PASSCODE

export const SHOW_PUSH_NOTIFICATION_OPTION = APP?.SHOW_PUSH_NOTIFICATION_OPTION

export const SHOW_ROOM_FILTER = APP?.SHOW_ROOM_FILTER

export const SHOW_UPDATE_BANNER = APP?.SHOW_UPDATE_BANNER

export const SHOW_WALLET = APP?.SHOW_WALLET

export const TRANSPARENT_TAB_BAR = APP?.TRANSPARENT_TAB_BAR

export const IS_EXPERIMENTAL_BUILD = Config?.KEET_VARIANT === 'experimental'

export const IS_CHAT_CACHE_ENABLED = APP?.IS_CHAT_CACHE_ENABLED

export const IS_TIME_STATS_ENABLED = APP?.IS_TIME_STATS_ENABLED

export const SHOW_ADMIN_RESET_UI = APP?.SHOW_ADMIN_RESET_UI

export const SHOW_REMOVE_DEVICE = APP?.SHOW_REMOVE_DEVICE

export const SHOW_RENAME_DEVICE = APP?.SHOW_RENAME_DEVICE
