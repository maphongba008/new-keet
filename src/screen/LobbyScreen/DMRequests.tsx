import React, { useCallback } from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { useSelector } from 'react-redux'
import { useNavigation } from '@react-navigation/native'

import { getDmRequests } from '@holepunchto/keet-store/store/room'

import { ButtonBase } from 'component/Button'
import SvgIcon from 'component/SvgIcon'
import { createThemedStylesheet, useTheme } from 'component/theme'
import { APPIUM_IDs, appiumTestProps } from 'lib/appium'
import s, {
  DIRECTION_CODE,
  UI_SIZE_2,
  UI_SIZE_8,
  UI_SIZE_14,
  UI_SIZE_16,
} from 'lib/commonStyles'
import { SCREEN_DM_REQUESTS } from 'lib/navigation'

import { useStrings } from 'i18n/strings'

const DMRequests = () => {
  const styles = getStyles()
  const strings = useStrings()
  const theme = useTheme()
  const dmRequests = useSelector(getDmRequests)
  const navigation = useNavigation<any>()
  const navigateToDmRequests = useCallback(() => {
    navigation.navigate(SCREEN_DM_REQUESTS)
  }, [navigation])
  return (
    <ButtonBase
      {...appiumTestProps(APPIUM_IDs.lobby_dm)}
      style={styles.container}
      onPress={navigateToDmRequests}
    >
      <SvgIcon name="chats" />
      <Text style={styles.text}>{strings.chat.dm.dmRequests}</Text>
      <View style={styles.count}>
        <Text style={styles.countText}>{dmRequests.length}</Text>
      </View>
      <SvgIcon name="chevronRight" color={theme.color.grey_200} />
    </ButtonBase>
  )
}

const getStyles = createThemedStylesheet((theme) => {
  const styles = StyleSheet.create({
    container: {
      ...s.centerAlignedRow,
      borderBottomWidth: 1,
      borderColor: theme.color.grey_600,
      paddingHorizontal: UI_SIZE_16,
      paddingVertical: UI_SIZE_8,
    },
    count: {
      backgroundColor: theme.color.red_600,
      borderRadius: UI_SIZE_16,
      paddingHorizontal: UI_SIZE_8,
      paddingVertical: UI_SIZE_2,
    },
    countText: {
      ...theme.text.body,
      color: theme.color.blue_400,
      fontSize: UI_SIZE_14,
      writingDirection: DIRECTION_CODE,
    },
    text: {
      ...theme.text.body,
      flex: 1,
      fontSize: UI_SIZE_14,
      marginLeft: UI_SIZE_8,
      writingDirection: DIRECTION_CODE,
    },
  })
  return styles
})

export default DMRequests
