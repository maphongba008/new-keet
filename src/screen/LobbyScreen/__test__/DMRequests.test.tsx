import { fireEvent, renderHook, screen } from '@testing-library/react-native'
import { useNavigation } from '@react-navigation/native'

import { APPIUM_IDs } from 'lib/appium'
import { SCREEN_DM_REQUESTS } from 'lib/navigation'
import { renderWithProviders } from 'lib/testUtils'

import DMRequests from '../DMRequests'

jest.mock('lib/hooks/useIsFeatureSupported', () => ({
  useIsDmSupported: () => {
    return true
  },
}))

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn().mockReturnValue({
    navigate: jest.fn(),
  }),
}))
describe('DMRequests', () => {
  test('render correctly', async () => {
    const {
      result: { current },
    } = renderHook(() => useNavigation())
    const element = <DMRequests />
    renderWithProviders(element, {
      preloadedState: {
        room: {
          dm: {
            received: {
              requests: [
                {
                  seq: 0,
                  type: 0,
                  seen: false,
                  roomId: 'roomId',
                  senderId: 'senderId',
                  invitation: 'invitation',
                  message: 'message',
                  timestamp: 1,
                },
                {
                  seq: 1,
                  type: 0,
                  seen: false,
                  roomId: 'roomId1',
                  senderId: 'senderId1',
                  invitation: 'invitation1',
                  message: 'message',
                  timestamp: 2,
                },
              ],
            },
          },
        },
      },
    })
    expect(screen.toJSON()).toMatchSnapshot()
    // test navigate to DM Screen
    const btn = await screen.findByTestId(APPIUM_IDs.lobby_dm)
    fireEvent.press(btn)
    expect(current.navigate).toHaveBeenCalledWith(SCREEN_DM_REQUESTS)
  })
})
