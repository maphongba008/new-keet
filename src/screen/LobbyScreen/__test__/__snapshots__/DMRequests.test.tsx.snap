// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DMRequests render correctly 1`] = `
<View
  accessibilityLabel="lobby_dm"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "opacity": 1,
      },
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#222430",
        "flexDirection": "row",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
  testID="lobby_dm"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    color="transparent"
    fill="transparent"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {},
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    testID="svg-icon-chats"
    tintColor="transparent"
    vbHeight={20}
    vbWidth={20}
    width={20}
    xml="<svg fill="currentColor" height="20" viewBox="0 0 20 20" width="20" xmlns="http://www.w3.org/2000/svg"><path d="m18.1075 14.6349c.4408-.896.6604-1.8846.6406-2.883s-.2786-1.97744-.7546-2.85525c-.476-.8778-1.1554-1.62878-1.9813-2.19003s-1.7743-.91644-2.7657-1.03581c-.3288-.76647-.8076-1.45938-1.4084-2.0379-.6007-.57853-1.3112-1.03097-2.08945-1.33067-.77831-.2997-1.60872-.4406-2.44232-.41439s-1.65352.21899-2.41146.56699-1.43857.84417-2.00178 1.45929-.99761 1.33674-1.27761 2.12234c-.28.78561-.39994 1.61931-.35274 2.45198.0472.83268.26057 1.64745.62755 2.39645l-.59844 2.0953c-.0534.1874-.05574.3858-.00677.5744.04896.1887.14745.3608.28527.4986.13782.1379.30997.2364.49863.2853.18866.049.38698.0466.57443-.0068l2.09532-.5984c.63227.3104 1.31239.512 2.01171.5961.33336.781.82238 1.4859 1.4372 2.0716.61483.5858 1.34257 1.0401 2.13886 1.3352.7962.2952 1.6442.425 2.4923.3815s1.6784-.2594 2.4402-.6344l2.0953.5984c.1875.0534.3858.0558.5745.0068.1886-.0489.3608-.1474.4986-.2853.1378-.1378.2363-.3099.2853-.4986.049-.1886.0466-.387-.0068-.5744zm-13.30159-2.1953c-.05808 0-.1159.0078-.17187.0234l-2.06328.5898.58984-2.0632c.04421-.1572.02514-.3255-.05312-.4688-.57468-1.05505-.74993-2.28185-.49363-3.45561.25629-1.17375.92697-2.21585 1.88912-2.93532.96216-.71948 2.15135-1.06816 3.34968-.98214s2.32555.60098 3.17505 1.4505c.8495.84953 1.3645 1.97673 1.4505 3.17506s-.2626 2.38751-.9821 3.34971c-.7195.9621-1.7616 1.6328-2.93535 1.8891-1.17376.2563-2.40056.081-3.45562-.4936-.09147-.0512-.19441-.0783-.29922-.0789zm12.03129 2.3.5898 2.0632-2.0633-.5898c-.1572-.0442-.3254-.0251-.4687.0531-1.1521.6268-2.5043.7763-3.7655.4163-1.26117-.36-2.33065-1.2007-2.97827-2.3413.85582-.0891 1.68399-.3541 2.43257-.7785.7485-.4243 1.4013-.9987 1.9174-1.6872.516-.6886.8842-1.4763 1.0815-2.31384.1972-.83754.2192-1.7068.0646-2.55325.7489.17649 1.447.52346 2.0399 1.01387.5929.49042 1.0646 1.11105 1.3784 1.81359.3139.70253.4613 1.46803.4309 2.23683-.0304.7689-.2379 1.5203-.6062 2.1959-.0789.1439-.098.3131-.0531.4711z" fill="#fff"/></svg>
"
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 0,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="m18.1075 14.6349c.4408-.896.6604-1.8846.6406-2.883s-.2786-1.97744-.7546-2.85525c-.476-.8778-1.1554-1.62878-1.9813-2.19003s-1.7743-.91644-2.7657-1.03581c-.3288-.76647-.8076-1.45938-1.4084-2.0379-.6007-.57853-1.3112-1.03097-2.08945-1.33067-.77831-.2997-1.60872-.4406-2.44232-.41439s-1.65352.21899-2.41146.56699-1.43857.84417-2.00178 1.45929-.99761 1.33674-1.27761 2.12234c-.28.78561-.39994 1.61931-.35274 2.45198.0472.83268.26057 1.64745.62755 2.39645l-.59844 2.0953c-.0534.1874-.05574.3858-.00677.5744.04896.1887.14745.3608.28527.4986.13782.1379.30997.2364.49863.2853.18866.049.38698.0466.57443-.0068l2.09532-.5984c.63227.3104 1.31239.512 2.01171.5961.33336.781.82238 1.4859 1.4372 2.0716.61483.5858 1.34257 1.0401 2.13886 1.3352.7962.2952 1.6442.425 2.4923.3815s1.6784-.2594 2.4402-.6344l2.0953.5984c.1875.0534.3858.0558.5745.0068.1886-.0489.3608-.1474.4986-.2853.1378-.1378.2363-.3099.2853-.4986.049-.1886.0466-.387-.0068-.5744zm-13.30159-2.1953c-.05808 0-.1159.0078-.17187.0234l-2.06328.5898.58984-2.0632c.04421-.1572.02514-.3255-.05312-.4688-.57468-1.05505-.74993-2.28185-.49363-3.45561.25629-1.17375.92697-2.21585 1.88912-2.93532.96216-.71948 2.15135-1.06816 3.34968-.98214s2.32555.60098 3.17505 1.4505c.8495.84953 1.3645 1.97673 1.4505 3.17506s-.2626 2.38751-.9821 3.34971c-.7195.9621-1.7616 1.6328-2.93535 1.8891-1.17376.2563-2.40056.081-3.45562-.4936-.09147-.0512-.19441-.0783-.29922-.0789zm12.03129 2.3.5898 2.0632-2.0633-.5898c-.1572-.0442-.3254-.0251-.4687.0531-1.1521.6268-2.5043.7763-3.7655.4163-1.26117-.36-2.33065-1.2007-2.97827-2.3413.85582-.0891 1.68399-.3541 2.43257-.7785.7485-.4243 1.4013-.9987 1.9174-1.6872.516-.6886.8842-1.4763 1.0815-2.31384.1972-.83754.2192-1.7068.0646-2.55325.7489.17649 1.447.52346 2.0399 1.01387.5929.49042 1.0646 1.11105 1.3784 1.81359.3139.70253.4613 1.46803.4309 2.23683-.0304.7689-.2379 1.5203-.6062 2.1959-.0789.1439-.098.3131-.0531.4711z"
        fill={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#ffffff",
        "flex": 1,
        "fontFamily": "OpenSans-Regular",
        "fontSize": 14,
        "includeFontPadding": false,
        "marginLeft": 8,
        "writingDirection": "ltr",
      }
    }
  >
    DM requests
  </Text>
  <View
    style={
      {
        "backgroundColor": "#CE3B3B",
        "borderRadius": 16,
        "paddingHorizontal": 8,
        "paddingVertical": 2,
      }
    }
  >
    <Text
      style={
        {
          "color": "#26D2E8",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "includeFontPadding": false,
          "writingDirection": "ltr",
        }
      }
    >
      2
    </Text>
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    color="#A0A8B8"
    fill="#A0A8B8"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {},
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    testID="svg-icon-chevronRight"
    tintColor="#A0A8B8"
    vbHeight={24}
    vbWidth={24}
    width={24}
    xml="<svg fill="currentColor" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"/></svg>
"
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4288719032,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="m17.031 12.531-7.49996 7.5c-.06968.0697-.15241.125-.24345.1627-.09105.0377-.18863.0571-.28717.0571-.09855 0-.19613-.0194-.28718-.0571-.09104-.0377-.17377-.093-.24345-.1627-.06968-.0696-.12496-.1524-.16267-.2434-.03771-.0911-.05712-.1886-.05712-.2872 0-.0985.01941-.1961.05712-.2872.03771-.091.09299-.1737.16267-.2434l6.97031-6.9694-6.97031-6.96936c-.14073-.14073-.21979-.3316-.21979-.53062 0-.19903.07906-.3899.21979-.53063s.3316-.21979.53063-.21979c.19902 0 .38989.07906.53062.21979l7.49996 7.50001c.0698.0696.1251.1524.1628.2434.0378.0911.0572.1887.0572.2872 0 .0986-.0194.1962-.0572.2872-.0377.0911-.093.1738-.1628.2434z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
